source/command-monitoring/tests/unified/redacted-commands.yml

```diff
@@ -177,76 +210,133 @@ tests:
               command: { copydb: { $$exists: false } }
 
   - description: "hello with speculative authenticate"
+    runOnRequirements:
+      - minServerVersion: "4.9"
     operations:
       - name: runCommand
         object: *database
         arguments:
           commandName: hello
           command:
-            hello: "private"
-            speculativeAuthenticate: "foo"
-        expectError:
-          isError: true
+            hello: 1
+            speculativeAuthenticate:
+              saslStart: 1
+    expectEvents:
+      - client: *client
+        events:
+          - commandStartedEvent:
+              commandName: hello
+              command:
+                hello: { $$exists: false }
+                speculativeAuthenticate: { $$exists: false }
+          - commandSucceededEvent:
+              commandName: hello
+              reply:
+                # We expect the reply not to be redacted, as the server can't
+                # successfully authenticate us, which means the reply does not
+                # need to be redacted. However, drivers may still choose to
+                # redact the reply based on the fact that the command contained
+                # sensitive information
+                isWritablePrimary: { $$unsetOrMatches: true }
+                # This assertion will currently always hold true since we're
+                # not expecting successful authentication, in which case this
+                # field is missing anyways.
+                speculativeAuthenticate: { $$exists: false }
```

Comment:
Even if I remove the logic for redacting replies, this test (and the corresponding one for legacy hello) still passes in the Go driver. The reply still exists (because I'm no longer redacting), `isWritablePrimary` is true, and `speculativeAuthenticate` does not exist because authentication failed.

Reply:
Yes, that is unfortunately true. I can either remove the reply assertions in this test or leave them in to show intent.

I've tried to create a user beforehand so we can actually test the auth process (and thus get a speculativeAuthenticate in the reply) but wasn't able to get it to pass consistently without spending more time. With the tight timeline on this I opted to take a more pragmatic approach.